controllers/common.go

```diff
@@ -522,25 +522,84 @@ func updateVSRFromRestore(vsr *volsnapmoverv1alpha1.VolumeSnapshotRestore, clien
 	return nil
 }
 
-func updateVSRStatusPhase(vsr *volsnapmoverv1alpha1.VolumeSnapshotRestore, phase volsnapmoverv1alpha1.VolumeSnapshotRestorePhase, client client.Client) error {
-	if vsr == nil {
-		return errors.New("nil vsr in updateVSRStatusPhase")
+func (r *VolumeSnapshotRestoreReconciler) updateVSRStatusPhase(repDest *volsyncv1alpha1.ReplicationDestination, phase volsnapmoverv1alpha1.VolumeSnapshotRestorePhase, client client.Client) error {
+	vsr := volsnapmoverv1alpha1.VolumeSnapshotRestore{}
+	if err := r.Get(r.Context, r.req.NamespacedName, &vsr); err != nil {
+		r.Log.Error(err, fmt.Sprintf("unable to fetch volumesnapshotrestore %s", r.req.NamespacedName))
+		return err
 	}
 
 	vsr.Status.Phase = phase
 
```

Comment:
@eemcmullan don't we need a check here to see if the VSR phase is `completed` ?

Reply:
Wait, I mean this function is called at other places as well, right ? Lets say you are calling this function for updated the VSR to `SnapshotBackupDone` or `InProgress`  phase, this function will update the completion ts in that case, right ?